File: website/client/src/utils/rooms.js
export const ROOM_IDS = ['living', 'kitchen', 'bedroom', 'entrance', 'bathroom', 'garden'];

export const ROOM_LABELS = {
    living: 'Living Room',
    kitchen: 'Kitchen',
    bedroom: 'Bedroom',
    entrance: 'Entrance',
    bathroom: 'Bathroom',
    garden: 'Garden',
};

export const DASHBOARD_ROOM_TABS = [
    { id: 'all', label: 'All' },
    { id: 'living', label: 'Living' },
    { id: 'kitchen', label: 'Kitchen' },
    { id: 'bedroom', label: 'Bedroom' },
    { id: 'entrance', label: 'Entrance' },
    { id: 'bathroom', label: 'Bathroom' },
    { id: 'garden', label: 'Garden' },
];

function deviceToRoom(deviceName) {
    const n = (deviceName || '').toLowerCase();
    if (n.includes('kitchen')) return 'kitchen';
    if (n.includes('bedroom')) return 'bedroom';
    if (n.includes('door') || n.includes('front') || n.includes('entrance')) return 'entrance';
    if (n.includes('bath')) return 'bathroom';
    if (n.includes('garden')) return 'garden';
    if (n.includes('living') || n.includes('main') || n.includes('rpi')) return 'living';
    return 'living';
}

export function resolveRoom(device) {
    if (!device) return 'living';
    if (device.room && ROOM_IDS.includes(device.room)) return device.room;
    return deviceToRoom(device.name);
}

export function matchesRoomTab(roomId, tabId) {
    if (tabId === 'all') return true;
    return roomId === tabId;
}

File: Mobile-app/lib/utils/rooms.dart
class DashboardRoomTab {
  final String id;
  final String label;

  const DashboardRoomTab(this.id, this.label);
}

const roomIds = <String>[
  'living',
  'kitchen',
  'bedroom',
  'entrance',
  'bathroom',
  'garden',
];

const roomLabels = <String, String>{
  'living': 'Living Room',
  'kitchen': 'Kitchen',
  'bedroom': 'Bedroom',
  'entrance': 'Entrance',
  'bathroom': 'Bathroom',
  'garden': 'Garden',
};

const dashboardRoomTabs = <DashboardRoomTab>[
  DashboardRoomTab('all', 'All'),
  DashboardRoomTab('living', 'Living'),
  DashboardRoomTab('kitchen', 'Kitchen'),
  DashboardRoomTab('bedroom', 'Bedroom'),
  DashboardRoomTab('entrance', 'Entrance'),
  DashboardRoomTab('bathroom', 'Bathroom'),
  DashboardRoomTab('garden', 'Garden'),
];

String deviceToRoom(String? deviceName) {
  final n = (deviceName ?? '').toLowerCase();
  if (n.contains('kitchen')) return 'kitchen';
  if (n.contains('bedroom')) return 'bedroom';
  if (n.contains('door') || n.contains('front') || n.contains('entrance')) {
    return 'entrance';
  }
  if (n.contains('bath')) return 'bathroom';
  if (n.contains('garden')) return 'garden';
  if (n.contains('living') || n.contains('main') || n.contains('rpi')) {
    return 'living';
  }
  return 'living';
}

String resolveRoom(Map<String, dynamic>? device) {
  if (device == null) return 'living';
  final room = device['room']?.toString();
  if (room != null && roomIds.contains(room)) return room;
  return deviceToRoom(device['name']?.toString());
}

bool matchesRoomTab(String roomId, String tabId) {
  if (tabId == 'all') return true;
  return roomId == tabId;
}

String roomLabel(String roomId) => roomLabels[roomId] ?? 'Home';
